common/scala/src/main/scala/whisk/common/Config.scala

```diff
@@ -39,20 +39,26 @@ package whisk.common
  *
  * @param propertiesFile a file which defines the properties
  */
-class Config(requiredProperties: Map[String, String]) {
+class Config(requiredProperties: Map[String, String], optionalProperties: Set[String] = Set()) extends Logging {
 
     def isValid: Boolean = valid
 
     /**
      * Gets value for key if it exists else the empty string.
+     * The value of the override key will instead be returned if its value is present in the map.
      *
      * @param key to lookup
+     * @param overrideKey the property whose value will be returned if the map contains the override key.
      * @return value for the key or the empty string if the key does not have a value/does not exist
      */
-    def apply(key: String): String = try {
-        settings(key)
+    def apply(key: String, overrideKey: String = ""): String = try {
```

Comment:
Is this the same as

`Try(settings(overrideKey)).orElse(Try(settings(key)).getOrElse("")` ?